pkg/controller/statusmanager/status_manager.go

```diff
@@ -379,8 +382,29 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...operv
 			)
 		}
 
-		if oc.Spec.DefaultNetwork.Type == operv1.NetworkTypeOpenShiftSDN {
-			// OpenShiftSDN is removed in 4.17, so block the upgrade if we have OpenShiftSDN in the spec.
+		netCfg := &configv1.Network{}
+		if err := status.client.ClientFor("").CRClient().
+			Get(context.Background(), types.NamespacedName{Name: names.OPERATOR_CONFIG}, netCfg); err != nil {
+			if !apierrors.IsNotFound(err) {
```

Comment:
Why not return for other kinds of error?

Reply:
if it's catchall, returning for all errors, the UTs break since this code path gets executed and there is not really even a cluster available.

do you want me to return for all errors and figure out some fix for the UTs?

Reply:
Yes, as long as it doesn't cost too much effort.

Reply:
@pliurh I just updated this to account for your review. can you PTAL when you have a chance?